ludwig/features/audio_feature.py

```diff
@@ -0,0 +1,301 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import logging
+import os
+
+import numpy as np
+import tensorflow as tf
+import soundfile
+
+from ludwig.constants import AUDIO, BACKFILL
+from ludwig.features.base_feature import BaseFeature
+from ludwig.features.sequence_feature import SequenceInputFeature
+
+from ludwig.utils.image_utils import get_abs_path
+from ludwig.utils.audio_utils import get_max_length_stft_based
+from ludwig.utils.audio_utils import calculate_incr_var
+from ludwig.utils.audio_utils import calculate_incr_mean
+from ludwig.utils.audio_utils import get_length_in_samp
+from ludwig.utils.audio_utils import get_group_delay
+from ludwig.utils.audio_utils import get_phase_stft_magnitude
+from ludwig.utils.audio_utils import get_stft_magnitude
+from ludwig.utils.audio_utils import get_non_symmetric_length
+from ludwig.utils.misc import set_default_value
+
+
+logger = logging.getLogger(__name__)
+
+
+class AudioBaseFeature(BaseFeature):
+    def __init__(self, feature):
+        super().__init__(feature)
+        self.type = AUDIO
+
+    preprocessing_defaults = {
+        'audio_file_length_limit_in_s': 7.5,
+        'missing_value_strategy': BACKFILL,
+        'in_memory': True,
+        'padding_value': 0,
+        'norm': None,
+        'audio_feature': {
+           'type': 'raw',
+        }
+    }
+
+    @staticmethod
+    def get_feature_meta(column, preprocessing_parameters):
+        audio_feature_dict = preprocessing_parameters['audio_feature']
+        first_audio_file_path = column[0]
+        _, sampling_rate_in_hz = soundfile.read(first_audio_file_path)
+
+        feature_dim = AudioBaseFeature._get_feature_dim(audio_feature_dict, sampling_rate_in_hz)
+        audio_file_length_limit_in_s = preprocessing_parameters['audio_file_length_limit_in_s']
+        max_length = AudioBaseFeature._get_max_length_feature(audio_feature_dict, sampling_rate_in_hz, audio_file_length_limit_in_s)
+        return { 
+                'feature_dim': feature_dim,
+                'sampling_rate_in_hz': sampling_rate_in_hz,
+                'max_length': max_length
+               }
+
+    @staticmethod
+    def _get_feature_dim(audio_feature_dict, sampling_rate_in_hz):
+        feature_type = audio_feature_dict['type']
+
+        if(feature_type == 'raw'):
```

Comment:
it's a style thing, but can you please use in all the file the if condition: convention instead of if(condition): convention and only use brackets if the condition is multiline (and always put a space before the condition or the parenthesis) please? the same applies to elif

Reply:
Also create constants for `raw`, `stft_phase`, etc.

Reply:
7ba4a6c